apps/api/src/app/workflows-v2/workflow.controller.ts

```diff
@@ -63,6 +67,23 @@ export class WorkflowController {
     );
   }
 
+  @Put(':workflowId/promote')
+  @ExternalApiAccessible()
+  @UseGuards(UserAuthGuard)
+  async promote(
+    @UserSession() user: UserSessionData,
+    @Param('workflowId') workflowId: string,
+    @Body() promoteWorkflowDto: PromoteWorkflowDto
+  ): Promise<WorkflowResponseDto> {
+    return this.syncToEnvironmentUseCase.execute(
+      SyncToEnvironmentCommand.create({
+        identifierOrInternalId: workflowId,
+        targetEnvironmentId: promoteWorkflowDto.targetEnvironmentId,
```

Comment:
I could wrap this usecase into `PromoteWorkflowUsecase` where I would retrieve environment with name "Production" to avoid the need for user to provide `targetEnvironmentId`. But I think this is a bit more flexible and we don't need additional usecase.

Reply:
i think we will have only Prod to promote to in the near-far future but as long as the API client knows the prod env i think this approach is good enough and as you said more flexible let's keep it for now.